plugin/test/unit/helpers/draft.config.spec.ts

```diff
@@ -0,0 +1,198 @@
+jest.mock('uuid')
+
+// TODO: Need proper TS types here!
+const mockReadJSON = readJSON as jest.MockedFunction<any>
+const mockEnsureFileSync = ensureFileSync as jest.MockedFunction<any>
+const mockCopySync = copySync as jest.MockedFunction<any>
+const mockWriteJSON = writeJSON as jest.MockedFunction<any>
+const mockReaddir = readdir as jest.MockedFunction<any>
+const mockUuidv4 = uuidv4 as jest.MockedFunction<any>
+
+/* eslint-disable max-nested-callbacks */
+/* eslint-disable max-lines-per-function */
+// Only added tests for 'createMetadataFileAndCopyDatastore'
+describe('createMetadataFileAndCopyDatastore', () => {
+  const mockedFileName = 'fake/path/mockFilename.mdb'
+  const mockDataMetdataJsonFile = {
+    fileName: mockedFileName,
+  }
+  const publishDir = resolve('public')
+  const cacheDir = resolve('.cache')
+
+  afterEach(() => {
+    mockReadJSON.mockRestore()
+    mockEnsureFileSync.mockRestore()
+    mockCopySync.mockRestore()
+    mockWriteJSON.mockRestore()
+    mockReaddir.mockRestore()
+    mockUuidv4.mockRestore()
+  })
+
+  describe('when previous filename exists in cached metadata file', () => {
+    beforeEach(() => {
+      mockReadJSON.mockResolvedValue(mockDataMetdataJsonFile)
+    })
+
+    it('should create expected filename in published directory', async () => {
+      const expectedFileName = `${publishDir}/${mockedFileName}`
+      await createMetadataFileAndCopyDatastore(publishDir, cacheDir)
+
+      expect(mockEnsureFileSync).toHaveBeenCalledWith(expectedFileName)
+    })
+
+    it('should copy cached datastore file to published directory', async () => {
+      const expectedDataFile = join(`${cacheDir}/data/datastore/data.mdb`)
+      const expectedFileName = `${publishDir}/${mockedFileName}`
+
+      await createMetadataFileAndCopyDatastore(publishDir, cacheDir)
+
+      expect(mockCopySync).toHaveBeenCalledWith(
+        expectedDataFile,
+        expectedFileName,
+      )
+    })
+
+    it('should create expected metadata file in cached directory', async () => {
+      const expectedFileName = `${cacheDir}/dataMetadata.json`
+
+      await createMetadataFileAndCopyDatastore(publishDir, cacheDir)
+
+      expect(mockEnsureFileSync).toHaveBeenCalledWith(expectedFileName)
+    })
+
+    it('should write expected metadata payload in cached directory', async () => {
+      const expectedFileName = `${cacheDir}/dataMetadata.json`
+      const expectedPayload = {
+        fileName: mockedFileName,
+      }
+
+      await createMetadataFileAndCopyDatastore(publishDir, cacheDir)
+
+      expect(mockWriteJSON).toHaveBeenCalledWith(
+        expectedFileName,
+        expectedPayload,
+      )
+    })
+  })
+
+  describe('when previous filename datastore file already exists in published directory', () => {
+    const mockedDatastoreFile = 'data-test-uuid.mdb'
+    const mockFilesInDirectory = ['testFile.mdb', mockedDatastoreFile]
+
+    beforeEach(() => {
+      mockReadJSON.mockResolvedValue(null)
```

Comment:
Explicitly mocking the `readJSON` function to `null` here. This could be done in a `beforeEach` in the top `describe` block if necessary but I was being explicit for the behaviour in this specific `describe` block.

Reply:
I think if the file doesn't exist, the `readJson` method will throw an error - so I imagine mocking rejection is what we'd want to do instead?

Reply:
Fixed.